Realm/Realm.Fody/ModuleWeaver.cs

```diff
@@ -497,6 +497,10 @@ private WeaveResult WeaveProperty(PropertyDefinition prop, TypeDefinition type,
                               new GenericInstanceMethod(_references.RealmObject_GetListValue) { GenericArguments = { elementType } },
                               concreteListConstructor);
         }
+        else if (prop.PropertyType.GetElementType().FullName == "System.Collections.Generic.List`1")
```

Comment:
I would move this check toward the end - perhaps before the `SetMethod == null` check. The motivation is primarily to keep "legit" cases toward the top and erroneous ones toward the bottom.